hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -1697,6 +1721,31 @@ int SaraNcpClient::modemPowerOff() {
     return 0;
 }
 
+int SaraNcpClient::modemSoftPowerOff() {
+    if (modemPowerState()) {
+        LOG(TRACE, "Try powering modem off using AT command");
+        if (ready_) {
+            int r = CHECK_PARSER(parser_.execCommand("AT+CPWROFF"));
+            if (r == AtResponse::OK) {
+                LOG(TRACE, "Modem is powered off");
+                // Disable voltage translator
+                modemSetUartState(false);
+                // WARN: We assume that the modem can turn off itself reliably.
+                ncpPowerState(NcpPowerState::OFF);
+            } else {
+                LOG(TRACE, "AT+CPWROFF command is not echoed");
```

Comment:
Echo in the context of AT commands is something different. Let's make this clearer. No response?